pysoa/common/schemas.py

```diff
@@ -34,3 +36,73 @@ def __repr__(self):
                 class_name=self.object_type.__name__,
             ) if self.object_type else '',
         )
+
+
+class PolymorphClassSchema(fields.Polymorph):
+    def __init__(self, contents_map, enforce_object_type_subclass_of=None, description=None, **kwargs):
+        super(PolymorphClassSchema, self).__init__(
+            switch_field='path',
+            contents_map=contents_map,
+            description=description,
+            **kwargs
+        )
+
+        self.contents_map = self.DynamicContentsMap(enforce_object_type_subclass_of, self.contents_map)
+
+    class DynamicContentsMap(dict):
```

Comment:
This feels like something you would want to use again.  Suggest making it a public class.

Reply:
GitHub PR comments are, frustratingly, not clear on the line on which you're actually commenting. To be clear, are you suggesting `DynamicContentsMap` be made a public class? If so, it doesn't really make sense to ever use this class on its own; it's necessarily very coupled with `PolymorphClassSchema`. With some considerable generifying, it could be extracted, but that seems like premature optimization absent a specific use case.

Reply:
reasonable